graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -812,10 +813,86 @@ def _preprocess_graphql_string(graphql_string):
     return graphql_string + '\n'
 
 
+def _sorted_tuple(unordered_list):
+    """Sort a list and turn it into a tuple.
+
+    Args:
+        unordered_list: list of strings
+
+    Returns:
+        tuple of the sorted input list
+    """
+    return tuple(sorted(unordered_list))
+
+
+def _validate_schema_and_ast(schema, ast):
+    """Validate the supplied graphql schema and ast.
+
+    This method wraps around graphql-core's validation to enforce a stricter requirement of the
+    schema-- all directives supported by the compiler must be declared by the schema, regardless of
+    whether each directive is used in the query or not.
+
+    Args:
+        schema: GraphQL schema object, created using the GraphQL library
+        ast: abstract syntax tree representation of a graphql query
+
+    Returns:
+        list containing schema and/or query validation errors
+    """
+    core_graphql_errors = validate(schema, ast)
+
+    # The following directives do not appear in the core-graphql library, but are permissable.
+    whitelisted_directives = set([
```

Comment:
I actually don't think we should allow any of these. Let's expect and allow only the ones defined by the compiler for now, until we have a better story of how custom directives fit into the picture.

Reply:
Some of the documentation is off, but I think we can keep the structure the same, idea being that we ensure that `parsed_schema_directives` = `expected_directives` + `unsupported_default_core-graphql_directives`